From 07230598d775c03d76f725a79857449120433c6a Mon Sep 17 00:00:00 2001
From: Zhang Ying-22455 <ying.zhang22455@nxp.com>
Date: Fri, 25 May 2018 14:51:11 +0800
Subject: [PATCH 523/767] Documentation: dt: binding: fsl: add devicetree
 binding for describing ftm alarm

The patch adds device tree probe support for fsl ftm-alarm driver.

Signed-off-by: Zhang Ying-22455 <ying.zhang22455@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../bindings/soc/fsl/layerscape/ftm-alarm.txt | 32 +++++++++++++++++++
 1 file changed, 32 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/soc/fsl/layerscape/ftm-alarm.txt

diff --git a/Documentation/devicetree/bindings/soc/fsl/layerscape/ftm-alarm.txt b/Documentation/devicetree/bindings/soc/fsl/layerscape/ftm-alarm.txt
new file mode 100644
index 000000000000..44c33b82b0d2
--- /dev/null
+++ b/Documentation/devicetree/bindings/soc/fsl/layerscape/ftm-alarm.txt
@@ -0,0 +1,32 @@
+Freescale FlexTimer Module (FTM) Alarm
+
+Required properties:
+
+- compatible : Should be "fsl,ftm-alarm" or "fsl,<chip>-ftm-alarm", the
+	       supported chips include
+	       "fsl,ls1012a-ftm-alarm"
+	       "fsl,ls1021a-ftm-alarm"
+	       "fsl,ls1043a-ftm-alarm"
+	       "fsl,ls1046a-ftm-alarm"
+	       "fsl,ls1088a-ftm-alarm"
+	       "fsl,ls208xa-ftm-alarm"
+- reg : Specifies base physical address and size of the register sets for the
+  FlexTimer Module and base physical address of IP Powerdown Exception Control
+  Register.
+- reg-names: names of the mapped memory regions listed in regs property.
+  should include the following entries:
+  "ftm":    Address of the register sets for FlexTimer Module
+  "pmctrl": Address of IP Powerdown Exception Control register
+- interrupts : Should be the FlexTimer Module interrupt.
+- big-endian: If the host controller is big-endian mode, specify this property.
+  The default endian mode is little-endian.
+
+Example:
+ftm0: ftm0@29d0000 {
+	compatible = "fsl,ls1043a-ftm-alarm";
+	reg = <0x0 0x29d0000 0x0 0x10000>,
+	      <0x0 0x1ee2140 0x0 0x4>;
+	reg-names = "ftm", "pmctrl";
+	interrupts = <0 86 0x4>;
+	big-endian;
+};
-- 
2.17.0

